src/services/expenses.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+let expenses = [];
+
+const init = () => {
+  expenses = [];
+};
+
+const getAll = (category, from, to) => {
+  let filteredExpenses = expenses;
+
+  if (category) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.category === category);
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses
+      .filter(expense => expense.spentAt >= from && expense.spentAt <= to);
+  }
+
+  return filteredExpenses;
+};
+
+const getById = (expenseId) => {
+  const foundExpense = expenses.find(expense => expense.id === +expenseId);
+
+  return foundExpense || null;
+};
+
+const create = ({ userId, spentAt, title, amount, category, note }) => {
+  const newExpense = {
+    id: expenses.length + 1,
```

Comment:
If you have previously removed some expenses, this code will cause you to have duplicate ids, which is not acceptable. You can keep track of your last id in a separate variable.

Same applies to users.